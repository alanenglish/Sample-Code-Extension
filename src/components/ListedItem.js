import React from 'react';
import PropTypes from 'prop-types';

const ListedItem = ({ listing, size }) => (
  <a href={`/listings/${listing.id}`}>
    <li className={`list-inline-item listed-item ${size}`}>
      <img
        className={`listed-item__img ${size}`}
        alt={listing.title}
        src={listing.photos[0]._links.small_crop.href}
      />
      <div className={`listed-item__title ${size}`}>
        {listing.make} - {listing.model}
      </div>
      <div className={`listed-item__price ${size}`}>
        {listing.price.display}
      </div>
    </li>
  </a>
);

ListedItem.propTypes = {
  listing: PropTypes.object,
  size: PropTypes.string
};

ListedItem.defaultProps = {
  listing: null,
  size: ''
};

export default ListedItem;
