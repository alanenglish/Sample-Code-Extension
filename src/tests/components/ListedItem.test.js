import React from 'react';
import { shallow } from 'enzyme';
import ListedItem from '../../components/ListedItem';
/* eslint-env jest */

test('should render ListedItem correctly', () => {
  const listing = {
    id: '123abc456def',
    title: 'Fender Bender',
    make: 'Fender',
    model: 'Bender',
    price: { display: '$1500' },
    photos: [{ _links: { small_crop: { href: 'http://www.alan.com' } } }]
  };

  const size = 'small';

  const wrapper = shallow(<ListedItem listing={listing} size={size} />);
  expect(wrapper).toMatchSnapshot();
});
