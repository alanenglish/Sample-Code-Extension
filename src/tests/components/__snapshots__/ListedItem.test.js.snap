// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ListedItem correctly 1`] = `
<a
  href="/listings/123abc456def"
>
  <li
    className="list-inline-item listed-item small"
  >
    <img
      alt="Fender Bender"
      className="listed-item__img small"
      src="http://www.alan.com"
    />
    <div
      className="listed-item__title small"
    >
      Fender
       - 
      Bender
    </div>
    <div
      className="listed-item__price small"
    >
      $1500
    </div>
  </li>
</a>
`;
